### ISSUE TITLE
False positives of DuplicateFareRuleZoneIdFieldsNotice when analyzing fare_rules.contains_id

### ISSUE THREAD

**Issue by evansiroky** [MobilityData/gtfs-validator#1144]
## **Bug report** 

The GTFS Validator is generating false positives of the DuplicateFareRuleZoneIdFieldsNotice for certain conditions involving the `fare_rules.txt#contains_id` field.

**Describe the bug**

When a fare rule is described that involves combinations of the `fare_rules.txt#contains_id`, the GTFS Validator generates false positives of the DuplicateFareRuleZoneIdFieldsNotice when a `fare_rules.txt#contains_id` is used correctly twice - especially when the `fare_rules.txt#contains_id` is also associated with a `fare_rules.txt#route_id`.

**How we reproduce the bug**

Example feed: [Archive.zip](https://github.com/MobilityData/gtfs-validator/files/8673998/Archive.zip)

In this feed in the `fare_rules.txt` file, the following use case is shown:

```csv
fare_id,route_id,origin_id,destination_id,contains_id
TL9,800,,,E
TL9,800,,,W
L1_E,800,,,E
L1_W,800,,,W
```

According to the [GTFS Spec about the `fare_rules.txt#contains_id` field](https://gtfs.org/schedule/reference/#fare_rulestxt#:~:text=Identifies%20the%20zones%20that%20a%20rider%20will%20enter%20while%20using%20a%20given%20fare%20class), it shows an example of a fare rule that applies only when all zones have been traversed. Therefore, in order to model a fare_rule that occurred in only one of the zones, one would expect there to be a "duplicate-looking" entry of another fare_rule that didn't have the same set of `contains_id` values as another fare rule.

Therefore, in the above use case, fare rule `TL9` applies only when travel is involved in all of zones `E` and `W`, but not in just `E` or just `W`. If travel occurs in just zone `E`, then fare rule `L1_E` applies. If travel occurs in just zone `W`, then fare rule `L1_W` applies. Additionally, all of that only applies to only route_id `800`.

**Expected behaviour**

The gtfs-validator should check the fare rules and compile a set of `contains_id` + `route_id` values when checking for duplicate fare rules.

**Observed behaviour**

The GTFS Validator flagged the above-mentioned use case as an error.

**Screenshots:** 
See relevant JSON output for the above-linked feed.

```json
{
  "code": "duplicate_fare_rule_zone_id_fields",
  "severity": "ERROR",
  "totalNotices": 2,
  "sampleNotices": [{
    "csvRowNumber": 4.0,
    "fareId": "L1_E",
    "previousCsvRowNumber": 2.0,
    "previousFareId": "TL9"
  }, {
    "csvRowNumber": 5.0,
    "fareId": "L1_W",
    "previousCsvRowNumber": 3.0,
    "previousFareId": "TL9"
  }]
}
```

**Environment versions**
- validator version: v2, v3
- Java version: 
- OS versions:

**Comment by KClough** [comment]
@evansiroky do you have a full GTFS data set that exhibits this behavior? 

It looks like the linked Archive.zip above only contains `fare_rules.txt` and `agency.txt`. 

With the current version of the validator, the report generated does not have `duplicate_fare_rule_zone_id_fields` errors.

**Comment by KClough** [comment]
@isabelle-dr after reviewing the documentation more closely, I think it is possible that this issue may no longer exist. I do not see any reference to `DuplicateFareRuleZoneIdFieldsNotice` outside of the migration docs.

**Comment by isabelle-dr** [comment]
@KClough this is included in the validator v4.0.0, but not in the latest version. 
<img width="1375" alt="Screenshot 2023-03-03 at 10 26 20 AM" src="https://user-images.githubusercontent.com/63653518/222759526-ff8b4753-37b8-47e6-b83b-2e7e211c09f9.png">

**Comment by isabelle-dr** [comment]
It was removed in this PR https://github.com/MobilityData/gtfs-validator/pull/1297

**Comment by isabelle-dr** [comment]
Okay, we had the same problem [in this issue](https://github.com/MobilityData/gtfs-validator/issues/1296#issuecomment-1324058037). It seems like this rule wasn't designed correctly. 

@evansiroky, this is fixed in the latest version of the validator, it will be updated for the next release. I am closing this issue.
